test/provider_os_memory.cpp

```diff
@@ -295,3 +297,48 @@ TEST_P(umfProviderTest, purge_force_INVALID_POINTER) {
     verify_last_native_error(provider.get(),
                              UMF_OS_RESULT_ERROR_PURGE_FORCE_FAILED);
 }
+
+// Tests for allocations on numa nodes
+TEST_F(test, checkNUMAnodesAllocation) {
+    if (numa_available() == -1 || numa_num_task_nodes() <= 1) {
+        GTEST_SKIP();
+    }
+
+    umf_memory_provider_handle_t os_memory_provider = nullptr;
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+
+    bitmask *nodemask = numa_parse_nodestring("1");
+    ASSERT_NE(nodemask, nullptr);
+    long unsigned int *nodemask_ptr = nodemask->maskp;
+
+    os_memory_provider_params.maxnode = numa_max_node() + 1;
+    os_memory_provider_params.nodemask = nodemask_ptr;
+    os_memory_provider_params.numa_mode = UMF_NUMA_MODE_BIND;
+
+    umf_result_t umf_result;
+    umf_result = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS,
+                                         &os_memory_provider_params,
+                                         &os_memory_provider);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(os_memory_provider, nullptr);
+
+    size_t size;
+    umf_result =
+        umfMemoryProviderGetMinPageSize(os_memory_provider, NULL, &size);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+    void *ptr = nullptr;
+    umf_result = umfMemoryProviderAlloc(os_memory_provider, size, 0, &ptr);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(ptr, nullptr);
+
+    int numa_node;
+    get_mempolicy(&numa_node, NULL, numa_max_node() + 1, (void *)ptr,
```

Comment:
Please check returned value. You can also create a wrapper `int get_numa_node(void* addr)` on top this function. In future will most likely replace `get_mempolicy` with a function from HWLOC.